python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import print_function
+import os
+import sys
+import socket
+import traceback
+import struct
+import cPickle
+import base64
+
+
+class SpecialLengths(object):
+    PYTHON_EXCEPTION_THROWN = -3
+    END_OF_STREAM = -4
+    NULL = -5
+
+
+class DataType(object):
+    INT = 1
+    LONG = 2
+    FLOAT = 3
+    DOUBLE = 4
+    BLOB = 5
+
+
+class Serializer(object):
+
+    @staticmethod
+    def read_long(stream):
+        obj = stream.read(8)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!q", obj)[0]
+
+    @staticmethod
+    def read_float(stream):
+        obj = stream.read(4)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!f", obj)[0]
+
+    @staticmethod
+    def read_double(stream):
+        obj = stream.read(8)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!d", obj)[0]
+
+    @staticmethod
+    def read_int(stream):
+        obj = stream.read(4)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!i", obj)[0]
+
+    @staticmethod
+    def write_int(value, stream):
+        stream.write(struct.pack("!i", value))
+
+    @staticmethod
+    def write_float(value, stream):
+        stream.write(struct.pack("!f", value))
+
+    @staticmethod
+    def write_double(value, stream):
+        stream.write(struct.pack("!d", value))
+
+    @staticmethod
+    def write_long(value, stream):
+        stream.write(struct.pack("!q", value))
+
+
+class PickleSerializer(Serializer):
+
+    @classmethod
```

Comment:
`cls` is the correct convention, RACO is wrong.   `@staticmethod` is best, if applicable.